src/basic/unit-name.h

```diff
@@ -16,6 +15,18 @@ typedef enum UnitNameFlags {
         _UNIT_NAME_INVALID = -EINVAL,
 } UnitNameFlags;
 
+#define UNIT_NAME_MAX_LONG    PATH_MAX-1
```

Comment:
if your intention to be able to stash any file system path into unit names, then PATH_MAX isnot going to work, since paths can basically contain almost any char, and we only allow non-crazy chars and expect the other ones to be escaped.